### PR TITLE
fix sonar issues for core application and entities

### PR DESCRIPTION
**Pull request by atomfrede**
To get our sonar rating forward this PR replaces https://github.com/jhipster/generator-jhipster/pull/10263

updates #10258

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/admin/logs/logs.component.html.ejs [hunk 1]
```diff
@@ -26,8 +26,8 @@
     <table class="table table-sm table-striped table-bordered" aria-describedby="logs-page-heading">
         <thead>
         <tr title="click to order">
-            <th (click)="orderProp = 'name'; reverse=!reverse"><span jhiTranslate="logs.table.name">Name</span></th>
-            <th (click)="orderProp = 'level'; reverse=!reverse"><span jhiTranslate="logs.table.level">Level</span></th>
+            <th scope="row" (click)="orderProp = 'name'; reverse=!reverse"><span jhiTranslate="logs.table.name">Name</span></th>
```
Comment (vishal423) [inline generators/client/templates/angular/src/main/webapp/app/admin/logs/logs.component.html.ejs]: shouldn't scope be `col` ?

File: generators/client/templates/angular/src/main/webapp/app/admin/tracker/tracker.component.html.ejs [hunk 2]
```diff
@@ -20,14 +20,14 @@
...
-        <table class="table table-striped">
+        <table class="table table-striped"  aria-describedby="tracker-page-heading">
             <thead>
                 <tr>
-                    <th jhiTranslate="tracker.table.userlogin">User</th>
-                    <th jhiTranslate="tracker.table.ipaddress">IP Address</th>
-                    <th jhiTranslate="tracker.table.page">Current page</th>
-                    <th jhiTranslate="tracker.table.time">Time</th>
-                    <th></th>
+                    <th scope="row" jhiTranslate="tracker.table.userlogin">User</th>
```
Comment (vishal423) [inline generators/client/templates/angular/src/main/webapp/app/admin/tracker/tracker.component.html.ejs]: shouldn't scope be `col` ?

File: generators/client/templates/angular/src/main/webapp/app/admin/user-management/user-management.component.html.ejs [hunk 3]
```diff
@@ -28,20 +28,20 @@
...
             <thead>
             <tr<% if (databaseType !== 'cassandra') { %> jhiSort [(predicate)]="predicate" [(ascending)]="reverse" [callback]="transition.bind(this)"<% } %>>
-                <th<% if (databaseType !== 'cassandra') { %> jhiSortBy="id"<% } %>><span jhiTranslate="global.field.id">ID</span><% if (databaseType !== 'cassandra') { %> <fa-icon [icon]="'sort'"></fa-icon><% } %></th>
-                <th<% if (databaseType !== 'cassandra') { %> jhiSortBy="login"<% } %>><span jhiTranslate="userManagement.login">Login</span><% if (databaseType !== 'cassandra') { %> <fa-icon [icon]="'sort'"></fa-icon><% } %></th>
-                <th<% if (databaseType !== 'cassandra') { %> jhiSortBy="email"<% } %>><span jhiTranslate="userManagement.email">Email</span><% if (databaseType !== 'cassandra') { %> <fa-icon [icon]="'sort'"></fa-icon><% } %></th>
-                <th></th>
+                <th scope="row" <% if (databaseType !== 'cassandra') { %> jhiSortBy="id"<% } %>><span jhiTranslate="global.field.id">ID</span><% if (databaseType !== 'cassandra') { %> <fa-icon [icon]="'sort'"></fa-icon><% } %></th>
```
Comment (vishal423) [inline generators/client/templates/angular/src/main/webapp/app/admin/user-management/user-management.component.html.ejs]: shouldn't scope be `col` ?

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management.component.html.ejs [hunk 4]
```diff
@@ -53,19 +53,19 @@
         <table class="table table-striped" aria-describedby="page-heading">
             <thead>
             <tr<% if (pagination !== 'no') { %> jhiSort [(predicate)]="predicate" [(ascending)]="reverse" [callback]="<%=pagination !== 'infinite-scroll' ? 'transition.bind(this)' : 'reset.bind(this)'%>"<% } %>>
-            <th<% if (pagination !== 'no') { %> jhiSortBy="id"<% } %>><span jhiTranslate="global.field.id">ID</span><% if (pagination !== 'no') { %> <fa-icon [icon]="'sort'"></fa-icon><% } %></th>
+            <th scope="row" <% if (pagination !== 'no') { %> jhiSortBy="id"<% } %>><span jhiTranslate="global.field.id">ID</span><% if (pagination !== 'no') { %> <fa-icon [icon]="'sort'"></fa-icon><% } %></th>
```
Comment (vishal423) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management.component.html.ejs]: shouldn't scope be `col` ?

File: generators/server/templates/src/main/resources/templates/error.html.ejs [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <!DOCTYPE html>
 <html xmlns:th="http://www.thymeleaf.org" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-      xsi:schemaLocation="http://www.thymeleaf.org ">
+      xsi:schemaLocation="http://www.thymeleaf.org" th:lang="${#locale.language}">
```
Comment (vishal423) [inline generators/server/templates/src/main/resources/templates/error.html.ejs]: are we pointing to user selected language?

File: generators/client/templates/angular/src/main/webapp/app/admin/logs/logs.component.html.ejs (hunk 1 above)
Comment (atomfrede) [reply generators/client/templates/angular/src/main/webapp/app/admin/logs/logs.component.html.ejs]: :man_facepalming: Yes of course, obviously it was too late yesterday

File: generators/server/templates/src/main/resources/templates/error.html.ejs (hunk 5 above)
Comment (atomfrede) [reply generators/server/templates/src/main/resources/templates/error.html.ejs]: No it seems to be always english/the default language at least it doesn't make any difference which language is selected or saved in the profile/

File: generators/client/templates/angular/src/main/webapp/app/admin/configuration/configuration.component.html.ejs [hunk 6]
```diff
@@ -43,12 +43,12 @@
         </tbody>
     </table>
     <div *ngFor="let key of keys(allConfiguration)">
-        <h4><span>{{key}}</span></h4>
-        <table class="table table-sm table-striped table-bordered table-responsive d-table">
+        <h4 [id]="key"><span>{{key}}</span></h4>
```
[the remark this answers is not in the document]
Comment (atomfrede) [reply generators/client/templates/angular/src/main/webapp/app/admin/configuration/configuration.component.html.ejs]: Using the index loop to make sure it is unique.

**Comment by vishal423** [comment]
thanks @atomfrede

**Comment by atomfrede** [comment]
Will have a look into the remaining more or less false positives (`th:lang` is not recognized as `lang` for example).

**Pull request merged**